x-pack/legacy/plugins/uptime/public/components/functional/monitor_status_details/__test__/monitor_ssl_certificate.test.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import moment from 'moment';
+import { mountWithIntl } from 'test_utils/enzyme_helpers';
+import { EuiBadge } from '@elastic/eui';
+import { renderWithIntl } from 'test_utils/enzyme_helpers';
+import { PingTls } from '../../../../../common/graphql/types';
+import { MonitorSSLCertificate } from '../monitor_status_bar';
+
+describe('MonitorStatusBar component', () => {
+  let monitorTls: PingTls;
+
+  beforeEach(() => {
+    const dateInTwoMonths = moment()
+      .add(2, 'month')
+      .toString();
+
+    monitorTls = {
+      certificate_not_valid_after: dateInTwoMonths,
+    };
+  });
+
+  it('renders', () => {
+    const component = renderWithIntl(<MonitorSSLCertificate tls={monitorTls} />);
+    expect(component).toMatchSnapshot();
+  });
+
+  it('renders null if invalid date', () => {
+    monitorTls = {
+      certificate_not_valid_after: 'i am so invalid date',
+    };
+    const component = renderWithIntl(<MonitorSSLCertificate tls={monitorTls} />);
+    expect(component).toMatchSnapshot();
+  });
+
+  it('renders expiring data if date is less than 30 days', () => {
+    const dateIn15Days = moment()
+      .add(15, 'day')
+      .toString();
+    monitorTls = {
+      certificate_not_valid_after: dateIn15Days,
+    };
+    const component = mountWithIntl(<MonitorSSLCertificate tls={monitorTls} />);
+
+    const badgeComponent = component.find(EuiBadge);
+    expect(badgeComponent.props().color).toBe('warning');
+
+    const badgeComponentText = component.find('.euiBadge__text');
+    expect(badgeComponentText.text()).toBe(moment(dateIn15Days).fromNow());
+
+    expect(badgeComponent.find('span.euiBadge--warning')).toBeTruthy();
+  });
+
+  it('renders expiring state if validity is greater than 30 days', () => {
```

Comment:
Did you mean to say "does not"? At any rate, I think a good name for this test would be "does not render the expiration date with a warning".

Reply:
done !!